qiskit/algorithms/evolvers/pvqd/pvqd.py

```diff
@@ -0,0 +1,390 @@
+            use_parameter_shift: If True, use the parameter shift rule to compute gradients.
+                If False, the optimizer will not be passed a gradient callable. In that case,
+                Qiskit optimizers will use a finite difference rule to approximate the gradients.
+            quantum_instance: The backend of quantum instance used to evaluate the circuits.
+        """
+        if evolution is None:
+            evolution = LieTrotter()
+
+        self.ansatz = ansatz
+        self.initial_parameters = initial_parameters
+        self.timestep = timestep
+        self.optimizer = optimizer
+        self.initial_guess = initial_guess
+        self.expectation = expectation
+        self.evolution = evolution
+        self.gradients = use_parameter_shift
+
+        self._sampler = None
+        self.quantum_instance = quantum_instance
+
+    @property
+    def quantum_instance(self) -> Optional[QuantumInstance]:
+        """Return the current quantum instance."""
+        return self._quantum_instance
+
+    @quantum_instance.setter
+    def quantum_instance(self, quantum_instance: Optional[Union[Backend, QuantumInstance]]) -> None:
+        """Set the quantum instance and circuit sampler."""
+        if quantum_instance is not None:
+            if not isinstance(quantum_instance, QuantumInstance):
+                quantum_instance = QuantumInstance(quantum_instance)
+            self._sampler = CircuitSampler(quantum_instance)
+
+        self._quantum_instance = quantum_instance
+
+    def step(
+        self,
+        hamiltonian: OperatorBase,
+        ansatz: QuantumCircuit,
+        theta: np.ndarray,
+        dt: float,
+        initial_guess: np.ndarray,
+    ) -> Tuple[np.ndarray, float]:
+        """Perform a single time step.
+
+        Args:
+            hamiltonian: The Hamiltonian under which to evolve.
+            ansatz: The parameterized quantum circuit which attempts to approximate the
+                time-evolved state.
+            theta: The current parameters.
+            dt: The time step.
+            initial_guess: The initial guess for the classical optimization of the
+                fidelity between the next variational state and the Trotter-evolved last state.
+                If None, this is set to a random vector with elements in the interval
+                :math:`[-0.01, 0.01]`.
+
+        Returns:
+            A tuple consisting of the next parameters and the fidelity of the optimization.
+        """
+        loss, gradient = self.get_loss(hamiltonian, ansatz, dt, theta)
+
+        if initial_guess is None:
+            initial_guess = np.random.random(self.initial_parameters.size) * 0.01
+
+        if isinstance(self.optimizer, Optimizer):
+            optimizer_result = self.optimizer.minimize(loss, initial_guess, gradient)
+        else:
+            optimizer_result = self.optimizer(loss, initial_guess, gradient)
```

Comment:
This is just a comment. I made it in VQE when we updating optimizer - had Optimizer implemented the callable interface there would be no need to make any test/check here and do things differently. Perhaps something to think about again going forwards.